flixel/graphics/tile/FlxTilesheet.hx

```diff
@@ -83,9 +84,148 @@ class FlxTilesheet extends Tileset
+		
+		var scaleIndex = 0;
+		var rotationIndex = 0;
+		var rgbIndex = 0;
+		var rgbOffsetIndex = 0;
+		var alphaIndex = 0;
+		var transformIndex = 0;
+		
+		var numValues = 3;
+		
+		if (useRect) { numValues = useOrigin ? 8 : 6; }
+		if (useScale) { scaleIndex = numValues; numValues ++; }
+		if (useRotation) { rotationIndex = numValues; numValues ++; }
+		if (useTransform) { transformIndex = numValues; numValues += 4; }
+		if (useRGB) { rgbIndex = numValues; numValues += 3; }
+		if (useAlpha) { alphaIndex = numValues; numValues ++; }
+		if (useRGBOffset) { rgbOffsetIndex = numValues; numValues += 4; }
+		
+		var totalCount = tileData.length;
+		if (count >= 0 && totalCount > count) totalCount = count;
+		var itemCount = Math.ceil (totalCount / numValues);
+		var iIndex = 0;
+		var tint = 0xFFFFFF;
+		
+		//var shader:GraphicsShader = shader != null ? cast shader : defaultShader;
+		var shader = new GraphicsShader ();
+		
+		shader.data.texture0.input = bitmapData;
+		shader.data.texture0.smoothing = smooth;
+		
+		if (useAlpha)
+		{
+			shader.data.alpha.value = [];
+		}
+		
+		if (useRGB)
+		{
+			shader.data.colorMultipliers.value = [];
+		}
+		
+		if (useRGBOffset)
+		{
+			shader.data.colorOffsets.value = [];
+		}
+		
+		var alphaValues = shader.data.alpha.value;
+		var colorMultiplierValues = shader.data.colorMultipliers.value;
+		var colorOffsetValues = shader.data.colorOffsets.value;
+		
+		for (i in 0...itemCount)
+		{
+			// useRect is always true
+			
+			rects.push (tileData[iIndex + 2]); //x
+			rects.push (tileData[iIndex + 3]); //y
+			rects.push (tileData[iIndex + 4]); //width
+			rects.push (tileData[iIndex + 5]); //height
+			
+			// useTransform is always true
+			
+			matrices.push (tileData[iIndex + transformIndex + 0]); //a
+			matrices.push (tileData[iIndex + transformIndex + 1]); //b
+			matrices.push (tileData[iIndex + transformIndex + 2]); //c
+			matrices.push (tileData[iIndex + transformIndex + 3]); //d
+			matrices.push (tileData[iIndex + 0]); //tx
+			matrices.push (tileData[iIndex + 1]); //ty
+			
+			// useAlpha is always true
+			
+			for (j in 0...6)
+			{
+				alphaValues.push (1);
+			}
+			
+			if (useRGB)
+			{
+				for (j in 0...6)
+				{
+					colorMultiplierValues.push (tileData[iIndex + rgbIndex]);
+					colorMultiplierValues.push (tileData[iIndex + rgbIndex + 1]);
+					colorMultiplierValues.push (tileData[iIndex + rgbIndex + 2]);
+					colorMultiplierValues.push (1);
+				}
+			}
+			
+			if (useRGBOffset)
+			{
+				for (j in 0...6)
+				{
+					colorOffsetValues.push (tileData[iIndex + rgbOffsetIndex]);
+					colorOffsetValues.push (tileData[iIndex + rgbOffsetIndex + 1]);
+					colorOffsetValues.push (tileData[iIndex + rgbOffsetIndex + 2]);
+					colorOffsetValues.push (tileData[iIndex + rgbOffsetIndex + 3]);
+				}
+			}
+			
+			iIndex += numValues;
+		}
+		
+		canvas.graphics.beginShaderFill (shader);
+		canvas.graphics.drawQuads (matrices, rects);
```

Comment:
I couldn't build for Flash on this line, with error: "flash.display.Graphics has no field drawQuads". Using latest OpenFL and lime develop branches.

Reply:
Thanks for taking a look, I just added a fix to OpenFL to fix compilation until we are able to finish the Flash compatibility implementation :smile:

Reply:
This is working almost out of the box for a couple of games I tried on native Windows and Flash (on Flash things appeared to work after just by commenting out that one line anyway). Will do some more testing.